lib/words_report.rb

```diff
@@ -0,0 +1,25 @@
+class WordsReport
+  attr_reader :words
+
+  # Arbitrary definition of popular
+  POPULAR_COUNT = 5
+
+  def initialize(counts)
+    @counts = counts
+    @words = words_by_frequency
```

Comment:
This call to `words_by_frequency` depending on the instance variable `@counts` being set first rubs me the wrong way, but I'm not sure my concern is legitimate. I would prefer passing `counts` as an argument.

Another way to write it to avoid my potentially illegitimate concern is to make `word_by_frequency` memoized and don't bother pre-calculating it. Even as I write this I can't see how that's better.